numerical_methods/false_position.cpp

```diff
@@ -19,56 +23,84 @@
 #include <iostream>
 #include <limits>
 
-#define EPSILON \
-    1e-6  // std::numeric_limits<double>::epsilon()  ///< system accuracy limit
-#define MAX_ITERATIONS 50000  ///< Maximum number of iterations to check
+int RANGE = 100000;  ///< Range in which we have to find the root. (-Range,Range)
+double INTERVAL_GAP = 0.5; /// interval gap. lesser the gap more the accuracy
 
 /** define \f$f(x)\f$ to find root for
  */
-static double eq(double i) {
-    return (std::pow(i, 3) - (4 * i) - 9);  // origial equation
+static double eq(double x) {
+    return (x*x-x);  // original equation
 }
 
-/** get the sign of any given number */
-template <typename T>
-int sgn(T val) {
-    return (T(0) < val) - (val < T(0));
+/**
+* This function finds root of the equation in given interval i.e. (x1,x2).
+* @param x1,x2 values for an interval in which root is present.
+  @param y1,y2 values of function at x1, x2 espectively.
+* @return root of the equation in the given interval.
+*/  
+static double regula_falsi(double x1,double x2,double y1,double y2){
+    double diff = x1-x2;
+    if(diff<0){
+        diff= (-1)*diff;
+    }
+    if(diff<0.00001){          
+        if(y1<0)
+        y1=-y1;
+        if(y2<0)
+        y2=-y2;
+        if(y1<y2)
+        return x1;
+        else
+        return x2;
+    }
+    double x3=0,y3=0;
+    x3 = x1 - (x1-x2)*(y1)/(y1-y2);
+    y3 = eq(x3);
+    return regula_falsi(x2,x3,y2,y3);
 }
 
-/** main function */
-int main() {
-    double a = -1, b = 1, x, z, m, n, c;
-    int i;
-
-    // loop to find initial intervals a, b
-    for (int i = 0; i < MAX_ITERATIONS; i++) {
-        z = eq(a);
-        x = eq(b);
-        if (sgn(z) == sgn(x)) {  // same signs, increase interval
-            b++;
-            a--;
-        } else {  // if opposite signs, we got our interval
-            break;
-        }
+/**
+* This function prints roots of the equation.
```

Comment:
```suggestion
* @brief This function prints roots of the equation.
```

Reply:
Done all requested changes.

Reply:
There was problem during batch commit. So, I did all the changes manually.